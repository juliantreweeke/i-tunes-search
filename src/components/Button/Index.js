import PropTypes from "prop-types";
import React from "react";
import LoadingSpinner from "../LoadingSpinner/Index";
import styles from "./button.module.css";

const Button = ({ children, variant, href, label, loading, ...props }) => {
  const HyperLinkButton = (
    <a
      href={href}
      aria-label={label}
      className={styles.button}
      data-variant={variant}
      target="_blank"
      rel="noreferrer"
    >
      {children}
    </a>
  );

  const StandardButton = (
    <button {...props} className={styles.button} data-variant={variant} data-loading={loading}>
      {loading && 
        <div className={styles.loadingSpinnerContainer}>
          <LoadingSpinner size="small" />
        </div>
        }
      {children}
    </button>
  );

  if (href) {
    return HyperLinkButton;
  }
  return StandardButton;
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Button;
