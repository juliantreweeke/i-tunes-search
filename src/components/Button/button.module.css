.button {
  background: var(--btn-primary-bg);
  font-weight: 700;
  cursor: pointer;
  display: inline-flex;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  line-height: 1;
  padding: var(--space-xs) var(--space-sm);
  border-radius: var(--radius);
  font-size: var(--text-md);
  color: var(--color-text-light);
  transition: .2s;
  border: none;
}

.button:hover {
  background-color: var(--btn-primary-hover);
}

.button:disabled{
  background: var(--btn-disabled-bg);
  cursor: not-allowed;
  box-shadow: none;
  opacity: 0.6;
} 