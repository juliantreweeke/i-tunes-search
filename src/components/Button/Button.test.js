import React from "react";
import { render } from "@testing-library/react";
import Button from "./Index";

describe("Button", () => {
  it("should render", () => {
    const { container } = render(<Button>Submit</Button>);
    expect(container).toBeTruthy();
  });

  it("Standard button matches snapshot with data props", () => {
    const { container } = render(<Button>Submit</Button>);
    expect(container).toMatchSnapshot();
  });

  it("Standard button matches snapshot with loading data props", () => {
    const { container } = render(<Button loading={true}>Submit</Button>);
    expect(container).toMatchSnapshot();
  });

  it("HyperLinkButton button matches snapshot with data props", () => {
    const { container } = render(
      <Button href="http://test.com" label="label">
        Buy now
      </Button>
    );
    expect(container).toMatchSnapshot();
  });
});
