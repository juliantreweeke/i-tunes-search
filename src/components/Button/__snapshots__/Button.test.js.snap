// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button HyperLinkButton button matches snapshot with data props 1`] = `
<div>
  <a
    aria-label="label"
    class="button"
    href="http://test.com"
    rel="noreferrer"
    target="_blank"
  >
    Buy now
  </a>
</div>
`;

exports[`Button Standard button matches snapshot with data props 1`] = `
<div>
  <button
    class="button"
  >
    Submit
  </button>
</div>
`;

exports[`Button Standard button matches snapshot with loading data props 1`] = `
<div>
  <button
    class="button"
    data-loading="true"
  >
    <div
      class="loadingSpinnerContainer"
    >
      <div
        class="spinner"
        data-size="small"
        id="loading-spinner"
      />
    </div>
    Submit
  </button>
</div>
`;
